server/src/test/java/org/elasticsearch/action/admin/indices/settings/put/UpdateSettingsRequestTests.java

```diff
@@ -107,4 +126,39 @@ private static Settings randomSettings(int min, int max) {
         return indices;
     }
 
+    private void doToFromXContentWithEnclosingSettingsField(boolean addSettingsField) throws IOException {
+        final UpdateSettingsRequest request = createTestInstance();
+        boolean humanReadable = randomBoolean();
+        final XContentType xContentType = randomFrom(XContentType.values());
+
+        BytesReference bytesRef;
+        if (addSettingsField) {
+            UpdateSettingsRequest requestWithEnclosingSettings = new UpdateSettingsRequest(request.settings()) {
+                public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+                    builder.startObject();
+                    builder.startObject("settings");
+                    this.settings().toXContent(builder, params);
+                    builder.endObject();
+                    builder.endObject();
+                    return builder;
+                }
+            };
+            BytesReference originalBytes = toShuffledXContent(requestWithEnclosingSettings, xContentType, ToXContent.EMPTY_PARAMS, humanReadable);
+            if (randomBoolean()) {
+                Predicate<String> excludeFilter = (s) -> s.startsWith("settings");
+                bytesRef = insertRandomFields(xContentType, originalBytes, excludeFilter, random());
```

Comment:
I noticed something strange in the `UpdateSettingsRequest#fromXContent` ( maybe not as part of this PR but it is easy to show it here ;) )

If the request is something like : 
```
{
  "field1" : "value1",
  "field2" : "value2",
  "settings" : 
  {
    "field3" : "value3"
  }
}
```
only `"field3" : "value3"` is parsed as a setting in the request ( exposed here by the `insertRandomFields` : inserting a random object at the top level is silently ignored )

Do we want such leniency in the requests ? 
Do we want to keep on supporting the possibility to wrap the settings in `"settings"` ?

Reply:
great catch, I would assume that we want to get rid of this but it's a potentially breaking change. Would you mind opening a new issue for this please?